diff --git a/Gemfile b/Gemfile
index 6d0921b..e04b34e 100644
--- a/Gemfile
+++ b/Gemfile
@@ -5,7 +5,6 @@ git_source(:github) do |repo_name|
   "https://github.com/#{repo_name}.git"
 end
 
-ruby '2.5.0'
 gem 'dotenv-rails', groups: [:development, :test]
 # Bundle edge Rails instead: gem 'rails', github: 'rails/rails'
 gem 'rails', '~> 5.1.6'
diff --git a/Gemfile.lock b/Gemfile.lock
index 2b20211..c20fd03 100644
--- a/Gemfile.lock
+++ b/Gemfile.lock
@@ -246,8 +246,5 @@ DEPENDENCIES
   uglifier (>= 1.3.0)
   web-console (>= 3.3.0)
 
-RUBY VERSION
-   ruby 2.5.0p0
-
 BUNDLED WITH
    1.16.1
diff --git a/app/.DS_Store b/app/.DS_Store
deleted file mode 100644
index 5008ddf..0000000
Binary files a/app/.DS_Store and /dev/null differ
diff --git a/app/assets/images/3 series.jpg b/app/assets/images/3 series.jpg
new file mode 100644
index 0000000..44a332c
Binary files /dev/null and b/app/assets/images/3 series.jpg differ
diff --git a/app/assets/images/a3.jpg b/app/assets/images/a3.jpg
new file mode 100644
index 0000000..8e6029f
Binary files /dev/null and b/app/assets/images/a3.jpg differ
diff --git a/app/assets/images/car.jpg b/app/assets/images/car.jpg
new file mode 100644
index 0000000..6e42f10
Binary files /dev/null and b/app/assets/images/car.jpg differ
diff --git a/app/assets/images/convertable.jpg b/app/assets/images/convertable.jpg
new file mode 100644
index 0000000..fa5f3b3
Binary files /dev/null and b/app/assets/images/convertable.jpg differ
diff --git a/app/assets/images/corvette.jpg b/app/assets/images/corvette.jpg
new file mode 100644
index 0000000..47fb8fe
Binary files /dev/null and b/app/assets/images/corvette.jpg differ
diff --git a/app/assets/images/cruize.jpeg b/app/assets/images/cruize.jpeg
new file mode 100644
index 0000000..f523a89
Binary files /dev/null and b/app/assets/images/cruize.jpeg differ
diff --git a/app/assets/images/elantra.jpg b/app/assets/images/elantra.jpg
new file mode 100644
index 0000000..322cd66
Binary files /dev/null and b/app/assets/images/elantra.jpg differ
diff --git a/app/assets/images/lancer.jpg b/app/assets/images/lancer.jpg
new file mode 100644
index 0000000..e516297
Binary files /dev/null and b/app/assets/images/lancer.jpg differ
diff --git a/app/assets/images/model 3.jpg b/app/assets/images/model 3.jpg
new file mode 100644
index 0000000..2980c3f
Binary files /dev/null and b/app/assets/images/model 3.jpg differ
diff --git a/app/assets/images/model s.jpg b/app/assets/images/model s.jpg
new file mode 100644
index 0000000..cea88fb
Binary files /dev/null and b/app/assets/images/model s.jpg differ
diff --git a/app/assets/images/model x.jpg b/app/assets/images/model x.jpg
new file mode 100644
index 0000000..27b69d9
Binary files /dev/null and b/app/assets/images/model x.jpg differ
diff --git a/app/assets/images/navara.jpg b/app/assets/images/navara.jpg
new file mode 100644
index 0000000..21ee854
Binary files /dev/null and b/app/assets/images/navara.jpg differ
diff --git a/app/assets/images/x3.jpg b/app/assets/images/x3.jpg
new file mode 100644
index 0000000..b1b0787
Binary files /dev/null and b/app/assets/images/x3.jpg differ
diff --git a/app/controllers/application_controller.rb b/app/controllers/application_controller.rb
index 9046faf..076b384 100644
--- a/app/controllers/application_controller.rb
+++ b/app/controllers/application_controller.rb
@@ -1,9 +1,9 @@
 class ApplicationController < ActionController::Base
-
+  before_action :authenticate_user!
   protect_from_forgery with: :exception
-  before_action :check_user_cars
+  before_action :check_user_for_owner_bookings
   before_action :check_user_for_guest_trips
-  before_action :set_user
+  # before_action :set_user
 
 
   private
@@ -11,9 +11,14 @@ class ApplicationController < ActionController::Base
     @user = current_user if user_signed_in?
   end
 
-  def check_user_cars
+  def check_user_for_owner_bookings
+    @bookings = false
     if user_signed_in?
-      @cars = current_user.cars
+      current_user.cars.each do |car|
+        if car.bookings.first
+          @bookings = true
+        end
+      end
     end
   end
 
diff --git a/app/controllers/bookings_controller.rb b/app/controllers/bookings_controller.rb
index ca04646..582ecbe 100644
--- a/app/controllers/bookings_controller.rb
+++ b/app/controllers/bookings_controller.rb
@@ -1,10 +1,6 @@
 class BookingsController < ApplicationController
 
   def new
-    unless user_signed_in?
-      redirect_to(new_user_registration_path)
-      flash[:notice] = "Nice choice! You'll have to sign in though before you book a car."
-    end
     @car = Car.find(params[:car_id])
     @days = params[:days]
 
@@ -14,21 +10,13 @@ class BookingsController < ApplicationController
     @booking = Booking.find(params[:id])
   end
 
-
-  def owner_cars
-    @cars = current_user.cars
-  end
-
-
   def owner_bookings
     @guest = false
     @active = false
-    @active2 = false
     @previous = false
+    @active2 = false
     @bookings = []
-    @cars = []
     current_user.cars.each do |car|
-      @cars.push(car)
       car.bookings.each do |booking|
         if booking.active
           @active = true
diff --git a/app/controllers/cars_controller.rb b/app/controllers/cars_controller.rb
index 5b29e91..5c85054 100644
--- a/app/controllers/cars_controller.rb
+++ b/app/controllers/cars_controller.rb
@@ -1,6 +1,6 @@
 class CarsController < ApplicationController
   before_action :set_car, only: [:show, :edit, :update, :destroy]
-  before_action :car_params, only: [:create, :destroy]
+  before_action :car_params, only: [:create, :edit, :update, :destroy]
 
 
   # GET /cars
@@ -29,16 +29,12 @@ class CarsController < ApplicationController
 
   # GET /cars/new
   def new
-    unless user_signed_in?
-      redirect_to(new_user_registration_path)
-      flash[:notice] = "Please sign up before you list a car"
-    end
+    redirect_to(new_user_registration_path) unless user_signed_in?
     @car = Car.new
   end
 
   # GET /cars/1/edit
   def edit
-    @car = Car.find(params[:id])
   end
 
   # POST /cars
@@ -62,26 +58,15 @@ class CarsController < ApplicationController
   # PATCH/PUT /cars/1
   # PATCH/PUT /cars/1.json
   def update
-    @car.update(make: params[:make], model: params[:model], year: params[:year],
-                full_address: params[:full_address], price_per_day: params[:price_per_day],
-                price_per_km: params[:price_per_km], image: params[:image])
-    redirect_to(bookings_owner_cars_path)
-    flash[:notice] = 'Car was successfully updated.'
-
-    # respond_to do |format|
-    #   if @car.update(car_params)
-    #     if params[:x]
-    #       format.html { redirect_to @car, notice: 'Car was successfully updated.' }
-    #       format.json { render :show, status: :ok, location: @car }
-    #     else
-    #       redirect_to(bookings_owner_cars_path)
-    #       flash[:notice] = 'Car was successfully updated.'
-    #     end
-    #   else
-    #     format.html { render :edit }
-    #     format.json { render json: @car.errors, status: :unprocessable_entity }
-    #   end
-    # end
+    respond_to do |format|
+      if @car.update(car_params)
+        format.html { redirect_to @car, notice: 'Car was successfully updated.' }
+        format.json { render :show, status: :ok, location: @car }
+      else
+        format.html { render :edit }
+        format.json { render json: @car.errors, status: :unprocessable_entity }
+      end
+    end
   end
 
   # DELETE /cars/1
diff --git a/app/mailers/booking_confirmation_mailer.rb b/app/mailers/booking_confirmation_mailer.rb
deleted file mode 100644
index 281c897..0000000
--- a/app/mailers/booking_confirmation_mailer.rb
+++ /dev/null
@@ -1,10 +0,0 @@
-class BookingConfirmationMailer < ApplicationMailer
-
-  default default: 'noreply@rentmycar.com'
-
-  def booking_confirmation_email(booking)
-    @booking = booking
-    mail(to: @booking.user.email, subject: 'Rent my car - Booking confirmation')
-  end
-
-end
diff --git a/app/mailers/user_notifier_mailer.rb b/app/mailers/user_notifier_mailer.rb
index 67a9103..a822af4 100644
--- a/app/mailers/user_notifier_mailer.rb
+++ b/app/mailers/user_notifier_mailer.rb
@@ -2,9 +2,9 @@ class UserNotifierMailer < ApplicationMailer
 
   default default: 'noreply@rentmycar.com'
 
-  def signup_email(user)
+  def send_signup_email(user)
     @user = user
-    mail(to: @user.email, subject: 'Rent my car - Signup confirmation')
+    mail(to: @user.email, subject: 'Thanks for signing up')
   end
 
 end
diff --git a/app/models/booking.rb b/app/models/booking.rb
index 96ecee6..d4da07e 100644
--- a/app/models/booking.rb
+++ b/app/models/booking.rb
@@ -1,11 +1,5 @@
 class Booking < ApplicationRecord
   belongs_to :user
   belongs_to :car
-
-  after_create :send_mail
-
-  def send_mail
-    BookingConfirmationMailer.booking_confirmation_email(self).deliver
-  end
-
+  
 end
diff --git a/app/models/car.rb b/app/models/car.rb
index 4541f5b..505eeca 100644
--- a/app/models/car.rb
+++ b/app/models/car.rb
@@ -6,7 +6,6 @@ class Car < ApplicationRecord
   resourcify
 
   geocoded_by :full_address
-  
   after_validation :geocode
 
   mount_uploader :image, CarImageUploader
diff --git a/app/models/user.rb b/app/models/user.rb
index 8665cf7..3787df9 100644
--- a/app/models/user.rb
+++ b/app/models/user.rb
@@ -10,8 +10,8 @@ class User < ApplicationRecord
   devise :database_authenticatable, :registerable,
          :recoverable, :rememberable, :trackable, :validatable
 
-  def send_mail
-    UserNotifierMailer.signup_email(self).deliver
-  end
+    def send_mail
+     UserNotifierMailer.send_signup_email(self).deliver
+   end
 
 end
diff --git a/app/views/booking_confirmation_mailer/booking_confirmation_email.html.erb b/app/views/booking_confirmation_mailer/booking_confirmation_email.html.erb
deleted file mode 100644
index b688757..0000000
--- a/app/views/booking_confirmation_mailer/booking_confirmation_email.html.erb
+++ /dev/null
@@ -1,48 +0,0 @@
-<!DOCTYPE html>
-<html lang="en" dir="ltr">
-  <head>
-    <meta charset="utf-8">
-    <title></title>
-  </head>
-  <body>
-
-    <h1>You're all set</h1>
-    <h3>Bookings details:</h3>
-
-    <table class="table mt-5">
-      <tbody>
-        <tr>
-          <td>You car</td>
-          <td><%= "#{@booking.car.make} #{@booking.car.model}" %></td>
-        </tr>
-        <tr>
-          <td>Owner</td>
-          <td><%= @booking.car.user.firstname %></td>
-        </tr>
-        <tr>
-          <td>Days</td>
-          <td><%= @booking.days %></td>
-        </tr> 
-        <tr>
-          <td>Price per day</td>
-          <td><%= number_to_currency(@booking.price_per_day) %></td>
-        </tr>
-        <tr style="border-top: solid grey 2px;">
-          <td class="d-flex justify-content-end pr-5">
-            <strong>Sub-total</strong>
-          </td>
-          <% t1 = @booking.price_per_day * @booking.days %>
-          <td><strong><%= number_to_currency(t1) %></strong><span class="text-success">PAID</span></td>
-        </tr>
-        <tr>
-          <td>Price per km</td>
-          <td><%= number_to_currency(@booking.price_per_km) %></td>
-        </tr>
-      </tbody>
-    </table>
-
-    <p>We'll email you a receipt for the final payment after your trip, pending kilometers travelled.<br>
-       </p>
-
-  </body>
-</html>
diff --git a/app/views/bookings/guest_trips.html.erb b/app/views/bookings/guest_trips.html.erb
index fe41caf..4ad1723 100644
--- a/app/views/bookings/guest_trips.html.erb
+++ b/app/views/bookings/guest_trips.html.erb
@@ -1,5 +1,4 @@
-<div class="ml-2" style="margin-top:100px;">
-
+<div class="m-2 m-lg-5">
   <div class="mb-5" style="max-width: 500px;">
 
     <% if @active %>
diff --git a/app/views/bookings/new.html.erb b/app/views/bookings/new.html.erb
index 1ee89d3..e710719 100644
--- a/app/views/bookings/new.html.erb
+++ b/app/views/bookings/new.html.erb
@@ -1,5 +1,5 @@
-<div class="ml-2" style="margin-top: 100px;">
-
+<div class="m-5">
+<br>
   <h3>Check your booking details</h3>
 
   <div class="row mt-5 d-flex justify-content-around">
@@ -9,7 +9,7 @@
         <span class="text-muted"><%= "#{@car.user.firstname.capitalize}'s" %></span>
         <span class="lead"><%= "#{@car.make.capitalize} #{@car.model.capitalize}" %></span>
       </div>
-        <%= image_tag @car.image_url, class: "img-thumbnail", style: "width: 400px;" %>
+        <%= image_tag "#{@car.model.downcase}.jpeg", class: "img-thumbnail", style: "width: 400px;" %>
     </div>
 
     <table class="table d-flex align-items-center" style="width:400px">
@@ -55,7 +55,6 @@
       </article>
       <%= hidden_field_tag :car_id, @car.id %>
       <%= hidden_field_tag :days, @days %>
-      <% end %>
 
       <script src="https://checkout.stripe.com/checkout.js" class="stripe-button"
               data-key="<%= Rails.configuration.stripe[:publishable_key] %>"
@@ -66,4 +65,11 @@
     </span>
   </div>
 
+  <% end %>
+
+  <br>
+  <br>
+
+
+
 </div>
diff --git a/app/views/bookings/owner_bookings.html.erb b/app/views/bookings/owner_bookings.html.erb
index 8bac1be..44a5fe1 100644
--- a/app/views/bookings/owner_bookings.html.erb
+++ b/app/views/bookings/owner_bookings.html.erb
@@ -1,6 +1,6 @@
-<div class="ml-2" style="margin-top:100px;">
-
+<div class="m-2 m-lg-5">
   <div class="mb-5" style="max-width: 500px;">
+
     <% if @active %>
       <h3>Your active bookings</h3>
       <% @bookings.each do |booking| %>
@@ -16,12 +16,16 @@
     <% end %>
 
     <% if @active2 %>
+
       <h3>Your previous bookings</h3>
+
       <% @bookings.each do |booking| %>
         <%= render 'non_active_bookings', {booking: booking, guest: @guest} %>
       <% end %>
+
       <%= link_to 'back', bookings_owner_bookings_path, class: "btn btn-dark", style: "color: white" %>
+
     <% end %>
-  </div>
 
+  </div>
 </div>
diff --git a/app/views/bookings/owner_cars.html.erb b/app/views/bookings/owner_cars.html.erb
deleted file mode 100644
index 69f058b..0000000
--- a/app/views/bookings/owner_cars.html.erb
+++ /dev/null
@@ -1,47 +0,0 @@
-<div class="ml-2" style="margin-top:100px;">
-
-  <h3>Your <%= "car".pluralize(@cars.count) %>:</h3>
-
-  <% current_user.cars.each do |car| %>
-
-    <div class="row mt-5 d-flex justify-content-around">
-
-
-      <div class="mb-5">
-        <div class="">
-          <span class="text-muted"><%= "#{current_user.firstname.capitalize}'s" %></span>
-          <span class="lead"><%= "#{car.make.capitalize} #{car.model.capitalize}" %></span>
-          <span class="text-muted"><small><%= car.year %></small></span>
-        </div>
-          <%= image_tag car.image_url, class: "img-thumbnail", style: "width: 400px;" %>
-      </div>
-
-      <table class="table d-flex align-items-center" style="width:400px">
-        <tbody>
-          <tr>
-            <th scope="row">Located at</th>
-            <td><%= car.full_address %></td>
-          </tr>
-          <tr>
-            <th scope="row">Price per day</th>
-            <td><%= number_to_currency(car.price_per_day) %></td>
-          </tr>
-          <tr>
-            <th scope="row">Price per km</th>
-            <td><%= number_to_currency(car.price_per_km) %></td>
-          </tr>
-          <tr>
-            <td scope="row"><%= link_to 'Edit', edit_car_path(car), class: "btn btn-dark", style: "color: white; width:90px;" %></td>
-            <td><%= link_to 'View bookings', bookings_owner_bookings_path, class: "btn btn-primary d-flex align-self-end", style: "color: white; width:145px;" %></td>
-          </tr>
-        </tbody>
-      </table>
-
-    </div>
-
-    <div class="row">
-    </div>
-
-  <% end %>
-
-</div>
diff --git a/app/views/cars/_form.html.erb b/app/views/cars/_form.html.erb
index 8b13789..ec5911c 100644
--- a/app/views/cars/_form.html.erb
+++ b/app/views/cars/_form.html.erb
@@ -1 +1,29 @@
+<%= form_with(model: car, local: true) do |form| %>
 
+  <% if car.errors.any? %>
+    <div id="error_explanation">
+      <h2><%= pluralize(car.errors.count, "error") %> prohibited this car from being saved:</h2>
+
+      <ul>
+      <% car.errors.full_messages.each do |message| %>
+        <li><%= message %></li>
+      <% end %>
+      </ul>
+    </div>
+  <% end %>
+
+  <div class="container d-flex justify-contents-center" style="width:200px;">
+    <div class="mt-5">
+
+      <%= form.text_field :make, placeholder: "Make", class: "mb-2 form-control form-control-sm"  %>
+      <%= form.text_field :model, placeholder: "Model", class: "mb-2 form-control form-control-sm"   %>
+      <%= form.number_field :year, placeholder: "Year", class: "mb-2 form-control form-control-sm"   %>
+      <%= form.text_field :full_address, placeholder: "Full Address", class: "mb-2 form-control form-control-sm"   %>
+      <%= form.text_field :price_per_day, placeholder: "Price per day", class: "mb-2 form-control form-control-sm"   %>
+      <%= form.text_field :price_per_km, placeholder: "Price per km", class: "mb-2 form-control form-control-sm"   %>
+      <%= form.file_field :image, class: "mb-2 form-control form-control-sm" %>
+      <%= form.submit "List it", class: "btn btn-sm btn-primary ml-2 form-inline" %>
+    </div>
+  </div>
+
+<% end %>
diff --git a/app/views/cars/edit.html.erb b/app/views/cars/edit.html.erb
index 00080d5..2a4d04f 100644
--- a/app/views/cars/edit.html.erb
+++ b/app/views/cars/edit.html.erb
@@ -1,67 +1,6 @@
-<div class="ml-2" style="margin-top:100px;">
+<h1>Editing Car</h1>
 
+<%= render 'form', car: @car %>
 
-    <h1>Edit car details</h1>
-
-    <%= form_tag car_path(@car), method: :patch, class: "form-inline d-flex justify-content-center" do %>
-
-      <table class="table m-5" style="max-width:800px;">
-        <tbody>
-          <tr>
-            <td><strong>Photo</strong></td>
-            <td class="text-center"><%= image_tag @car.image_url, class: "img-thumbnail", style: "max-width:200px;" %></td>
-            <td class="text-right"><%= file_field_tag :image %></td>
-          </tr>
-          <% if current_user.has_role? :admin %>
-           <% hidden_field_tag :x, true %>
-            <tr>
-              <td><strong>User_id</strong></td>
-              <td class="lead text-center"><%= @car.user.id %></td>
-            </tr>
-          <% end %>
-          <tr>
-            <td><strong>Make</strong></td>
-            <td class="lead text-center"><%= @car.make %></td>
-            <td class="text-right"><%= text_field_tag :make, @car.make, class: "form-control form-control-sm" %></td>
-          </tr>
-          <tr>
-            <td><strong>Model</strong></td>
-            <td class="lead text-center"><%= @car.model %></td>
-            <td class="text-right"><%= text_field_tag :model, @car.model, class: "form-control form-control-sm" %></td>
-          </tr>
-          <tr>
-            <td><strong>year</strong></td>
-            <td class="lead text-center"><%= @car.year %></td>
-            <td class="text-right"><%= text_field_tag :year, @car.year, class: "form-control form-control-sm" %></td>
-          </tr>
-          <tr>
-            <td><strong>Price per day</strong></td>
-            <td class="lead text-center"><%= number_to_currency(@car.price_per_day) %></td>
-            <td class="text-right"><%= text_field_tag :price_per_day, @car.price_per_day, class: "form-control form-control-sm" %></td>
-          </tr>
-          <tr>
-            <td><strong>Price per km</strong></td>
-            <td class="lead text-center"><%= number_to_currency(@car.price_per_km) %></td>
-            <td class="text-right"><%= text_field_tag :price_per_km, @car.price_per_km, class: "form-control form-control-sm" %></td>
-          </tr>
-          <tr>
-            <td><strong>Located at</strong></td>
-            <td class="text-center"><%= @car.full_address %></td>
-            <td class="text-right"><%= text_field_tag :full_address, @car.full_address, class: "form-control form-control-sm" %></td>
-          </tr>
-        </body>
-      </table>
-
-      <div class="d-flex justify-content-between" style="width: 80vw;">
-        <% if current_user.has_role? :admin %>
-          <%= link_to "Back", cars_path(@car), class: "btn btn-dark" %>
-        <% else %>
-          <%= link_to "Back", bookings_owner_cars_path, class: "btn btn-dark" %>
-        <% end %>
-          <%= submit_tag "Update", class: "btn btn-warning" %>
-      </div>
-
-
-    <% end %>
-
-</div>
+<%= link_to 'Show', @car %> |
+<%= link_to 'Back', cars_path %>
diff --git a/app/views/cars/index.html.erb b/app/views/cars/index.html.erb
index 159c174..464d3a7 100644
--- a/app/views/cars/index.html.erb
+++ b/app/views/cars/index.html.erb
@@ -1,132 +1,139 @@
-<div class="ml-2" style="margin-top:100px;">
-
-      <table>
-        <thead>
-          <tr>
-            <th></th>
-            <th>User</th>
-            <th>Make</th>
-            <th>Model</th>
-            <th>Year</th>
-            <th>Location</th>
-            <th>Price per day</th>
-            <th>Price per km</th>
-            <th colspan="3"></th>
-          </tr>
-        </thead>
-
-        <tbody>
-          <% @cars.each do |car| %>
-            <tr>
-              <td><%= image_tag car.image_url(:thumb) %></td>
-              <td><%= car.user.email %></td>
-              <td><%= car.make %></td>
-              <td><%= car.model %></td>
-              <td><%= car.year %></td>
-              <td><%= car.full_address %></td>
-              <td><%= car.price_per_day %></td>
-              <td><%= car.price_per_km %></td>
-              <td><%= link_to 'Show', car %></td>
-              <td><%= link_to 'Edit', edit_car_path(car) %></td>
-              <td><%= link_to 'Destroy', car, method: :delete, data: { confirm: 'Are you sure?' } %></td>
-            </tr>
-          <% end %>
-        </tbody>
-      </table>
-      <br>
-      <%= link_to 'New Car', new_car_path %>
-      <br>
-
-  <div class="m-2">
-    <%= form_tag cars_path, method: :get, class: "form-inline mb-3" do %>
-        <%= text_field_tag :user_location, '', placeholder: "Your location", class: "form-control", style: "width: 350px;" %>
-        <%= submit_tag "Go", class: "btn btn-primary ml-2", style: "width: 90px;" %>
-    <% end %>
-  </div>
-
-  <style>
-    /* Always set the map height explicitly to define the size of the div
-     * element that contains the map. */
-    #map {
-      height: 100%;
-    }
-    /* Optional: Makes the sample page fill the window. */
-    html, body {
-      height: 100%;
-      margin: 0;
-      padding: 0;
-    }
-  </style>
+<p id="notice"><%= notice %></p>
 
-  <div id="map"></div>
-    <script>
-      var map;
-      function initMap() {
+<% if current_user.has_role?(:admin) %>
+  <table>
+    <thead>
+      <tr>
+        <th></th>
+        <th>User</th>
+        <th>Make</th>
+        <th>Model</th>
+        <th>Year</th>
+        <th>Location</th>
+        <th>Price per day</th>
+        <th>Price per km</th>
+        <th colspan="3"></th>
+      </tr>
+    </thead>
 
-        map = new google.maps.Map(document.getElementById('map'), {
-          zoom: 10,
-          center: new google.maps.LatLng(<%= @latitude %>, <%= @longitude %>),
-          mapTypeId: 'roadmap'
-        });
+    <tbody>
+      <% @cars.each do |car| %>
+        <tr>
+          <td><%= image_tag car.image_url(:thumb) %></td>
+          <td><%= car.user.email %></td>
+          <td><%= car.make %></td>
+          <td><%= car.model %></td>
+          <td><%= car.year %></td>
+          <td><%= car.full_address %></td>
+          <td><%= car.price_per_day %></td>
+          <td><%= car.price_per_km %></td>
+          <td><%= link_to 'Show', car %></td>
+          <td><%= link_to 'Edit', edit_car_path(car) %></td>
+          <td><%= link_to 'Destroy', car, method: :delete, data: { confirm: 'Are you sure?' } %></td>
+        </tr>
+      <% end %>
+    </tbody>
+  </table>
+  <br>
+  <%= link_to 'New Car', new_car_path %>
+  <br>
+<% end %>
+
+<div class="m-2">
+  <%= form_tag cars_path, method: :get, class: "form-inline mb-3" do %>
+      <%= text_field_tag :user_location, '', placeholder: "Your location", class: "form-control", style: "width: 150px;" %>
+      <%= submit_tag "Go", class: "btn btn-primary ml-2", style: "width: 90px;" %>
+  <% end %>
+</div>
+
+<style>
+  /* Always set the map height explicitly to define the size of the div
+   * element that contains the map. */
+  #map {
+    height: 100%;
+  }
+  /* Optional: Makes the sample page fill the window. */
+  html, body {
+    height: 100%;
+    margin: 0;
+    padding: 0;
+  }
+</style>
+
+<div id="map"></div>
+  <script>
+
+
+
+
+    var map;
+    function initMap() {
+
+      map = new google.maps.Map(document.getElementById('map'), {
+        zoom: 10,
+        center: new google.maps.LatLng(<%= @latitude %>, <%= @longitude %>),
+        mapTypeId: 'roadmap'
+      });
 
-        var iconBase = 'https://maps.google.com/mapfiles/kml/shapes/';
-
-        var icons = {
-          info: {
-            icon: '/1x1.png'
-          }
-        };
-
-        var features = [
-          <% @cars.each do |car| %>
-          {
-            position: new google.maps.LatLng(<%= car.latitude %>, <%= car.longitude %>),
-            icon: 'info',
-            url: '/cars/<%=car.id%>',
-            image: '<%= car.image %>',
-            make: '<%= car.make %>',
-            model: '<%= car.model %>',
-            price_per_day: '<%= number_to_currency(car.price_per_day) %>',
-            price_per_km: '<%= number_to_currency(car.price_per_km) %>',
-
-          }
-          <% end %>
-        ];
-
-        features.forEach(function(feature) {
-
-          var contentString =
-            '<div class="d-flex info-window" onclick="window.location=\''+feature.url+'\'">'+
-              '<div class="mr-2">'+
-                '<span class="">        <img style="width:70px;" src='+feature.image+'>    </span>' +
+      var iconBase = 'https://maps.google.com/mapfiles/kml/shapes/';
+      var icons = {
+        info: {
+          icon: '/1x1.png'
+        }
+      };
+
+      var features = [
+        <% @cars.each do |car| %>
+        {
+          position: new google.maps.LatLng(<%= car.latitude %>, <%= car.longitude %>),
+          icon: 'info',
+          url: '/cars/<%=car.id%>',
+          image: '<%= car.image %>',
+          make: '<%= car.make %>',
+          model: '<%= car.model %>',
+          price_per_day: '<%= number_to_currency(car.price_per_day) %>',
+          price_per_km: '<%= number_to_currency(car.price_per_km) %>',
+
+        },
+        <% end %>
+      ];
+
+      // Create markers.
+      features.forEach(function(feature) {
+        var contentString =
+          '<div class="d-flex info-window" onclick="window.location=\''+feature.url+'\'">'+
+            '<div class="mr-2">'+
+              '<span class="">    <img style="width:70px;" src='+feature.image+'>    </span>' +
+            '</div>'+
+            '<div class="flex-columns">'+
+              '<div class="mb-1">'+
+                '<span class="">    <strong>'+feature.make+'</strong>     </span>'+
+                '<span class="">    <strong>'+feature.model+'</strong>     </span>'+
               '</div>'+
-              '<div class="flex-columns">'+
-                '<div class="mb-1">'+
-                  '<span class="">      <strong>'+feature.make+'</strong>                  </span>'+
-                  '<span class="">      <strong>'+feature.model+'</strong>                 </span>'+
-                '</div>'+
-                '<div class="text-success">'+
-                  '<span class="">      '+feature.price_per_day+' <span>/day</span>        </span>'+
-                  '<span class="ml-2">  '+feature.price_per_km+'  <span>/km</span>         </span>'+
-                '</div>'+
+              '<div class="text-success">'+
+                '<span class="">    '+feature.price_per_day+' <span>/day</span>   </span>'+
+                '<span class="ml-2">    '+feature.price_per_km+'  <span>/km</span>    </span>'+
               '</div>'+
-            '</div>';
-
-          var infowindow = new google.maps.InfoWindow({
-           content: contentString
-          });
-
-          var marker = new google.maps.Marker({
-            position: feature.position,
-            map: map,
-            icon: icons['info'].icon
-          });
-           infowindow.open(map, marker);
+            '</div>'+
+          '</div>';
+
+        var infowindow = new google.maps.InfoWindow({
+         content: contentString
+           });
+
+        // Can delete this now?
+        var marker = new google.maps.Marker({
+          position: feature.position,
+          map: map,
+          icon: icons['info'].icon,
+
+
         });
-      }
-    </script>
+
+         infowindow.open(map, marker);
+      });
+    }
+  </script>
   <script async defer
   src="https://maps.googleapis.com/maps/api/js?key=AIzaSyDSl2bo0vFAwrntL4R4GybQWzqHNSg8_9o&callback=initMap">
   </script>
-
-</div>
diff --git a/app/views/cars/new.html.erb b/app/views/cars/new.html.erb
index 8c9cb3b..890279e 100644
--- a/app/views/cars/new.html.erb
+++ b/app/views/cars/new.html.erb
@@ -1,34 +1,7 @@
-<div class="background-image" style="height:100vh;">
+<div class="background-image">
 
-  <div class="d-flex justify-content-center align-items-end" style="height:42%;">
-    <div class="mb-3" style="width:250px;">
-      <%= form_with(model: @car, local: true) do |form| %>
-        <% if @car.errors.any? %>
-          <div id="error_explanation">
-            <h2><%= pluralize(@car.errors.count, "error") %> prohibited this car from being saved:</h2>
-            <ul>
-            <% @car.errors.full_messages.each do |message| %>
-              <li><%= message %></li>
-            <% end %>
-            </ul>
-          </div>
-        <% end %>
-        <%= form.text_field :make, placeholder: "Make", class: "mb-2 form-control form-control-sm"  %>
-        <%= form.text_field :model, placeholder: "Model", class: "mb-2 form-control form-control-sm"   %>
-        <%= form.number_field :year, placeholder: "Year", class: "mb-2 form-control form-control-sm"   %>
-        <%= form.text_field :full_address, placeholder: "Full Address", class: "mb-2 form-control form-control-sm"   %>
-        <%= form.text_field :price_per_day, placeholder: "Price per day", class: "mb-2 form-control form-control-sm"   %>
-        <%= form.text_field :price_per_km, placeholder: "Price per km", class: "mb-2 form-control form-control-sm"   %>
-        <%= form.file_field :image, class: "mb-2 form-control form-control-sm" %>
-    </div>
-  </div>
+  <%= render 'form', car: @car %>
 
-  <div class="d-flex justify-content-center">
-    <div class="d-flex justify-content-between" style="width:250px;">
-      <%= link_to 'Back', cars_path, class: "btn btn-dark", style: "color: white; width: 90px;" %>
-      <%= form.submit "List it", class: "btn btn-primary form-inline", style: "color: white; width: 90px;" %>
-      <% end %>
-    </div>
-  </div>
+  <%= link_to 'Back', cars_path %>
 
 </div>
diff --git a/app/views/cars/show.html.erb b/app/views/cars/show.html.erb
index 8e059d1..e9a0c44 100644
--- a/app/views/cars/show.html.erb
+++ b/app/views/cars/show.html.erb
@@ -20,15 +20,15 @@
       <table class="table table-bordered">
         <tbody>
           <tr>
-            <td>Location</td>
+            <td>Location:</td>
             <td><%= @car.full_address %></td>
           </tr>
           <tr>
-            <td>Price per day</td>
+            <td>Price per day:</td>
             <td><%= number_to_currency(@car.price_per_day) %></td>
           </tr>
           <tr>
-            <td>Price per km</td>
+            <td>Price per km:</td>
             <td><%= number_to_currency(@car.price_per_km) %></td>
           </tr>
         </tbody>
diff --git a/app/views/devise/registrations/edit.html.erb b/app/views/devise/registrations/edit.html.erb
index 3ec8fd1..10ed32a 100644
--- a/app/views/devise/registrations/edit.html.erb
+++ b/app/views/devise/registrations/edit.html.erb
@@ -1,47 +1,43 @@
-<div class="container d-flex justify-content-center align-items-center" style="height:100vh;">
-  <div class="">
-    <h2>Edit <%= resource_name.to_s.humanize %></h2>
-
-    <%= form_for(resource, as: resource_name, url: registration_path(resource_name), html: { method: :put }) do |f| %>
-      <%= devise_error_messages! %>
-
-      <div class="field">
-        <%= f.label :email %><br />
-        <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
-      </div>
-
-      <% if devise_mapping.confirmable? && resource.pending_reconfirmation? %>
-        <div>Currently waiting confirmation for: <%= resource.unconfirmed_email %></div>
-      <% end %>
-
-      <div class="field">
-        <%= f.label :password %> <i>(leave blank if you don't want to change it)</i><br />
-        <%= f.password_field :password, autocomplete: "off" %>
-        <% if @minimum_password_length %>
-          <br />
-          <em><%= @minimum_password_length %> characters minimum</em>
-        <% end %>
-      </div>
-
-      <div class="field">
-        <%= f.label :password_confirmation %><br />
-        <%= f.password_field :password_confirmation, autocomplete: "off" %>
-      </div>
-
-      <div class="field">
-        <%= f.label :current_password %> <i>(we need your current password to confirm your changes)</i><br />
-        <%= f.password_field :current_password, autocomplete: "off" %>
-      </div>
-
-      <div class="actions">
-        <%= f.submit "Update" %>
-      </div>
+<h2>Edit <%= resource_name.to_s.humanize %></h2>
+
+<%= form_for(resource, as: resource_name, url: registration_path(resource_name), html: { method: :put }) do |f| %>
+  <%= devise_error_messages! %>
+
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+  </div>
+
+  <% if devise_mapping.confirmable? && resource.pending_reconfirmation? %>
+    <div>Currently waiting confirmation for: <%= resource.unconfirmed_email %></div>
+  <% end %>
+
+  <div class="field">
+    <%= f.label :password %> <i>(leave blank if you don't want to change it)</i><br />
+    <%= f.password_field :password, autocomplete: "off" %>
+    <% if @minimum_password_length %>
+      <br />
+      <em><%= @minimum_password_length %> characters minimum</em>
     <% end %>
+  </div>
 
-    <h3>Cancel my account</h3>
+  <div class="field">
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "off" %>
+  </div>
 
-    <p>Unhappy? <%= button_to "Cancel my account", registration_path(resource_name), data: { confirm: "Are you sure?" }, method: :delete %></p>
+  <div class="field">
+    <%= f.label :current_password %> <i>(we need your current password to confirm your changes)</i><br />
+    <%= f.password_field :current_password, autocomplete: "off" %>
+  </div>
 
-    <%= link_to "Back", :back %>
+  <div class="actions">
+    <%= f.submit "Update" %>
   </div>
-</div>
+<% end %>
+
+<h3>Cancel my account</h3>
+
+<p>Unhappy? <%= button_to "Cancel my account", registration_path(resource_name), data: { confirm: "Are you sure?" }, method: :delete %></p>
+
+<%= link_to "Back", :back %>
diff --git a/app/views/devise/registrations/new.html.erb b/app/views/devise/registrations/new.html.erb
index 5419b53..602803c 100644
--- a/app/views/devise/registrations/new.html.erb
+++ b/app/views/devise/registrations/new.html.erb
@@ -1,38 +1,29 @@
-<div class="container d-flex justify-content-center align-items-center" style="height:100vh;">
-  <div class="">
+<h2>Sign up</h2>
 
-    <p class="notice lead text-primary"><%= notice %></p>
+<%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
+  <%= devise_error_messages! %>
 
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+  </div>
 
-    <h2>Sign up</h2>
-
-    <%= form_for(resource, as: resource_name, url: registration_path(resource_name)) do |f| %>
-      <%= devise_error_messages! %>
-
-      <div class="field">
-        <%= f.label :email %><br />
-        <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
-      </div>
-
-      <div class="field">
-        <%= f.label :password %>
-        <% if @minimum_password_length %>
-        <em>(<%= @minimum_password_length %> characters minimum)</em>
-        <% end %><br />
-        <%= f.password_field :password, autocomplete: "off" %>
-      </div>
-
-      <div class="field">
-        <%= f.label :password_confirmation %><br />
-        <%= f.password_field :password_confirmation, autocomplete: "off" %>
-      </div>
-
-      <div class="actions">
-        <%= f.submit "Sign up" %>
-      </div>
-    <% end %>
+  <div class="field">
+    <%= f.label :password %>
+    <% if @minimum_password_length %>
+    <em>(<%= @minimum_password_length %> characters minimum)</em>
+    <% end %><br />
+    <%= f.password_field :password, autocomplete: "off" %>
+  </div>
 
-    <%= render "devise/shared/links" %>
+  <div class="field">
+    <%= f.label :password_confirmation %><br />
+    <%= f.password_field :password_confirmation, autocomplete: "off" %>
+  </div>
 
+  <div class="actions">
+    <%= f.submit "Sign up" %>
   </div>
-</div>
+<% end %>
+
+<%= render "devise/shared/links" %>
diff --git a/app/views/devise/sessions/new.html.erb b/app/views/devise/sessions/new.html.erb
index f3d2e80..3ebb001 100644
--- a/app/views/devise/sessions/new.html.erb
+++ b/app/views/devise/sessions/new.html.erb
@@ -1,30 +1,26 @@
-<div class="container d-flex justify-content-center align-items-center" style="height:100vh;">
-  <div class="">
-    <h2>Log in</h2>
+<h2>Log in</h2>
 
-    <%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
-      <div class="field">
-        <%= f.label :email %><br />
-        <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
-      </div>
-
-      <div class="field">
-        <%= f.label :password %><br />
-        <%= f.password_field :password, autocomplete: "off" %>
-      </div>
+<%= form_for(resource, as: resource_name, url: session_path(resource_name)) do |f| %>
+  <div class="field">
+    <%= f.label :email %><br />
+    <%= f.email_field :email, autofocus: true, autocomplete: "email" %>
+  </div>
 
-      <% if devise_mapping.rememberable? -%>
-        <div class="field">
-          <%= f.check_box :remember_me %>
-          <%= f.label :remember_me %>
-        </div>
-      <% end -%>
+  <div class="field">
+    <%= f.label :password %><br />
+    <%= f.password_field :password, autocomplete: "off" %>
+  </div>
 
-      <div class="actions">
-        <%= f.submit "Log in" %>
-      </div>
-    <% end %>
+  <% if devise_mapping.rememberable? -%>
+    <div class="field">
+      <%= f.check_box :remember_me %>
+      <%= f.label :remember_me %>
+    </div>
+  <% end -%>
 
-    <%= render "devise/shared/links" %>
+  <div class="actions">
+    <%= f.submit "Log in" %>
   </div>
-</div>
+<% end %>
+
+<%= render "devise/shared/links" %>
diff --git a/app/views/home/home.html.erb b/app/views/home/home.html.erb
index 402aba6..5ad6142 100644
--- a/app/views/home/home.html.erb
+++ b/app/views/home/home.html.erb
@@ -7,24 +7,18 @@
   <body>
 
 
-    <div class="background-image">
+    <div class="background-image pb-5">
       <div class="d-flex">
-        <div class="home-buttons container-fluid d-flex justify-content-around align-items-end mb-5"
+        <div class="home-buttons container-fluid d-flex justify-content-around align-items-start mt-5"
              style="height: 100vh">
           <%= link_to 'Find a car', cars_path,
             class: "btn btn-lg btn-info d-flex align-items-center justify-content-center mr-5",
             style: "width: 220px; height: 80px; font-size: 25px; color: white;" %>
-          <% if user_signed_in? %>
-            <% if current_user.cars.first %>
-              <%= link_to 'List another car', new_car_path,
-                class: "btn btn-lg btn-info d-flex align-items-center justify-content-center",
-                style: "width: 220px; height: 80px; font-size: 25px; color: white;" %>
-              <% else %>
-              <%= link_to 'List my car', new_car_path,
-                class: "btn btn-lg btn-info d-flex align-items-center justify-content-center",
-                style: "width: 220px; height: 80px; font-size: 25px; color: white;" %>
-            <% end %>
-          <% else %>
+          <% if current_user.cars %>
+            <%= link_to 'List another car', new_car_path,
+              class: "btn btn-lg btn-info d-flex align-items-center justify-content-center",
+              style: "width: 220px; height: 80px; font-size: 25px; color: white;" %>
+            <% else %>
             <%= link_to 'List my car', new_car_path,
               class: "btn btn-lg btn-info d-flex align-items-center justify-content-center",
               style: "width: 220px; height: 80px; font-size: 25px; color: white;" %>
diff --git a/app/views/layouts/application.html.erb b/app/views/layouts/application.html.erb
index 63d1777..b3b0b1c 100644
--- a/app/views/layouts/application.html.erb
+++ b/app/views/layouts/application.html.erb
@@ -34,9 +34,9 @@
               <a><%= link_to 'My trips', bookings_guest_trips_path, class: "nav-link", style: "color: white;" %></a>
             </li>
           <% end %>
-          <% if @cars %>
+          <% if @bookings %>
             <li class="nav-item mr-5">
-              <a><%= link_to 'My car'.pluralize(@cars.count), bookings_owner_bookings_path, class: "nav-link", style: "color: white;" %></a>
+              <a><%= link_to 'My bookings', bookings_owner_bookings_path, class: "nav-link", style: "color: white;" %></a>
             </li>
           <% end %>
           <% if user_signed_in? %>
@@ -55,6 +55,9 @@
       </div>
     </nav>
 
+    <p class="notice"><%= notice %></p>
+    <p class="alert"><%= alert %></p>
+
     <%= yield %>
 
     <script src="https://code.jquery.com/jquery-3.3.1.slim.min.js" integrity="sha384-q8i/X+965DzO0rT7abK41JStQIAqVgRVzpbzo5smXKp4YfRvH+8abtTE1Pi6jizo" crossorigin="anonymous"></script>
diff --git a/app/views/user_notifier_mailer/signup_email.html.erb b/app/views/user_notifier_mailer/send_signup_email.html.erb
similarity index 55%
rename from app/views/user_notifier_mailer/signup_email.html.erb
rename to app/views/user_notifier_mailer/send_signup_email.html.erb
index 1b069eb..05013e4 100644
--- a/app/views/user_notifier_mailer/signup_email.html.erb
+++ b/app/views/user_notifier_mailer/send_signup_email.html.erb
@@ -5,7 +5,8 @@
     <title></title>
   </head>
   <body>
-    <h1>Thanks for signing up!</h1>
-    <p>Your login email address is: <%= @user.email %></p>
+    <h1>Thanks for signing up to my website <%= @user.email %></h1>
+    <p>Now sign in and get started!</p>
   </body>
 </html>
+ 
diff --git a/config/database.yml b/config/database.yml
index 534aac6..880e0ae 100644
--- a/config/database.yml
+++ b/config/database.yml
@@ -41,7 +41,7 @@ development:
 
   # The TCP port the server listens on. Defaults to 5432.
   # If your server runs on a different port number, change accordingly.
-  port: 5432
+  #port: 5432
 
   # Schema search path. The server defaults to $user,public
   #schema_search_path: myapp,sharedapp,public
diff --git a/config/environments/production.rb b/config/environments/production.rb
index 36cac20..1a5176e 100644
--- a/config/environments/production.rb
+++ b/config/environments/production.rb
@@ -86,19 +86,6 @@ Rails.application.configure do
     config.logger    = ActiveSupport::TaggedLogging.new(logger)
   end
 
-
-
-    ActionMailer::Base.smtp_settings = {
-    :user_name => ENV['SENDGRID_USER'],
-    :password => ENV['SENDGRID_PASS'],
-    :domain => 'coderacademy.edu.au',
-    :address => 'smtp.sendgrid.net',
-    :port => 587,
-    :authentication => :plain,
-    :enable_starttls_auto => true
-  }
-
-
   # Do not dump schema after migrations.
   config.active_record.dump_schema_after_migration = false
 end
diff --git a/config/routes.rb b/config/routes.rb
index 713a06f..19c62d2 100644
--- a/config/routes.rb
+++ b/config/routes.rb
@@ -4,8 +4,6 @@ Rails.application.routes.draw do
 
   get 'bookings/show'
 
-  get 'bookings/owner_cars'
-
   get 'bookings/owner_bookings'
 
   get 'bookings/previous_bookings'
@@ -21,6 +19,8 @@ Rails.application.routes.draw do
 
   resources :cars
 
+  get 'cars/book'
+
   devise_for :users
 
   get 'home/home'
diff --git a/db/seeds.rb b/db/seeds.rb
index 6c45ae7..efcb508 100644
--- a/db/seeds.rb
+++ b/db/seeds.rb
@@ -1,8 +1,7 @@
 
-User.create([
-  {email: 'revsails@gmail.com', password: 'sparkle', firstname: 'Revel', lastname: 'Roestorf'},
+User.create([{email: 'revsails@gmail.com', password: 'sparkle', firstname: 'Revel', lastname: 'Roestorf'},
   {email: 'jane@gmail', password: 'sparkle', firstname: 'Jane', lastname: 'Jensen'},
-  {email: 'revel.roestorf@gmail.com', password: 'sparkle', firstname: 'Zac', lastname: 'Chop'},
+  {email: 'zac@gmail', password: 'sparkle', firstname: 'Zac', lastname: 'Chop'},
   {email: 'holly@gmail', password: 'sparkle', firstname: 'Holly', lastname: 'May'},
   {email: 'ray@gmail', password: 'sparkle', firstname: 'Ray', lastname: 'Charles'},
   {email: 'jay@gmail', password: 'sparkle', firstname: 'Jay', lastname: 'Bart'},
@@ -11,25 +10,24 @@ User.create([
   {email: 'butch@gmail', password: 'sparkle', firstname: 'Butch', lastname: 'Aspen'}
   ])
 
-Car.create([
-  {user_id: 1, image: "image/upload/v1525310148/eswn9zwqoocuvyzztr73.jpg", make: 'Tesla', model: 'model x',
-    year: '2016', full_address: '168 hedges ave, mermaid beach, qld', price_per_day: 90, price_per_km: 0.25},
-  {user_id: 1, image: "image/upload/v1525310084/bpu91s6qaf2lngmem6gc.jpg", make: 'Chevrolet', model: 'corvette',
-    year: '2018', full_address: '168 hedges ave, mermaid beach, qld', price_per_day: 80, price_per_km: 0.4},
-  {user_id: 2, image: "image/upload/v1525310097/zmbyia9pal1kkg0yey7d.jpg", make: 'Hyundai', model: 'elantra',
-    year: '2017', full_address: '1 breakfast creek rd, newstead, qld', price_per_day: 25, price_per_km: 0.18},
-  {user_id: 3, image: "image/upload/v1525310135/otmlg82qob3ftpkkjbyc.jpg", make: 'Tesla', model: 'model s',
-    year: '2015', full_address: '158 robinson st, fortitude valley, qld', price_per_day: 70, price_per_km: 0.2},
-  {user_id: 4, image: "image/upload/v1525310051/eespcwoezpang96kn7hf.jpg", make: 'Audi', model: 'a3',
-    year: '2006', full_address: '327 given terrace, paddington, qld', price_per_day: 32, price_per_km: 0.22},
-  {user_id: 5, image: "image/upload/v1525310168/zlkfkaoixuj2vt6w7nc2.jpg", make: 'BMW', model: 'x3',
-    year: '2008', full_address: '76 the esplanade, burleigh heads, qld', price_per_day: 30, price_per_km: 0.25},
-  {user_id: 6, image: "image/upload/v1525310067/bubuw6jorzdnp6fpfpol.jpg", make: 'Mini', model: 'convertable',
-    year: '2017', full_address: '30 pinnacle ct, robina, qld', price_per_day: 60, price_per_km: 0.35},
-  {user_id: 7, image: "image/upload/v1525308599/vgofl9azwghtfaoxycy6.jpg", make: 'Mitsubishi', model: 'lancer',
-    year: '2013', full_address: '26 lather st, southport, qld', price_per_day: 20, price_per_km: 0.2},
-  {user_id: 8, image: "image/upload/v1525310034/dpomfwsgiywnswrdkpfu.jpg", make: 'BMW', model: '3 series',
-    year: '2009', full_address: '8 beard st, auchenflower, qld', price_per_day: 40, price_per_km: 0.28},
-  {user_id: 9, image: "image/upload/v1525310119/c8fufkuthtj5itvtglks.jpg", make: 'Tesla', model: 'model 3',
-    year: '2018', full_address: '2 booran st, hope island, qld', price_per_day: 80, price_per_km: 0.15}
+Car.create([{user_id: 1, image: "image/upload/v1525310148/eswn9zwqoocuvyzztr73.jpg",
+  make: 'Tesla', model: 'model x', year: '2016', full_address: '168 hedges ave, mermaid beach, qld', price_per_day: 90, price_per_km: 0.25},
+  {user_id: 1, image: "image/upload/v1525310084/bpu91s6qaf2lngmem6gc.jpg",
+    make: 'Chevrolet', model: 'corvette', year: '2018', full_address: '168 hedges ave, mermaid beach, qld', price_per_day: 80, price_per_km: 0.4},
+  {user_id: 2, image: "image/upload/v1525310097/zmbyia9pal1kkg0yey7d.jpg",
+    make: 'Hyundai', model: 'elantra', year: '2017', full_address: '1 breakfast creek rd, newstead, qld', price_per_day: 25, price_per_km: 0.18},
+  {user_id: 3, image: "image/upload/v1525310135/otmlg82qob3ftpkkjbyc.jpg",
+    make: 'Tesla', model: 'model s', year: '2015', full_address: '158 robinson st, fortitude valley, qld', price_per_day: 70, price_per_km: 0.2},
+  {user_id: 4, image: "image/upload/v1525310051/eespcwoezpang96kn7hf.jpg",
+    make: 'Audi', model: 'a3', year: '2006', full_address: '327 given terrace, paddington, qld', price_per_day: 32, price_per_km: 0.22},
+  {user_id: 5, image: "image/upload/v1525310168/zlkfkaoixuj2vt6w7nc2.jpg",
+    make: 'BMW', model: 'x3', year: '2008', full_address: '76 the esplanade, burleigh heads, qld', price_per_day: 30, price_per_km: 0.25},
+  {user_id: 6, image: "image/upload/v1525310067/bubuw6jorzdnp6fpfpol.jpg",
+    make: 'Mini', model: 'convertable', year: '2017', full_address: '30 pinnacle ct, robina, qld', price_per_day: 60, price_per_km: 0.35},
+  {user_id: 7, image: "image/upload/v1525308599/vgofl9azwghtfaoxycy6.jpg",
+    make: 'Mitsubishi', model: 'lancer', year: '2013', full_address: '26 lather st, southport, qld', price_per_day: 20, price_per_km: 0.2},
+  {user_id: 8, image: "image/upload/v1525310034/dpomfwsgiywnswrdkpfu.jpg",
+    make: 'BMW', model: '3 series', year: '2009', full_address: '8 beard st, auchenflower, qld', price_per_day: 40, price_per_km: 0.28},
+  {user_id: 9, image: "image/upload/v1525310119/c8fufkuthtj5itvtglks.jpg",
+    make: 'Tesla', model: 'model 3', year: '2018', full_address: '2 booran st, hope island, qld', price_per_day: 80, price_per_km: 0.15}
   ])
diff --git a/test/mailers/booking_confirmation_mailer_test.rb b/test/mailers/booking_confirmation_mailer_test.rb
deleted file mode 100644
index 15bec4d..0000000
--- a/test/mailers/booking_confirmation_mailer_test.rb
+++ /dev/null
@@ -1,7 +0,0 @@
-require 'test_helper'
-
-class BookingConfirmationMailerTest < ActionMailer::TestCase
-  # test "the truth" do
-  #   assert true
-  # end
-end
diff --git a/test/mailers/previews/booking_confirmation_mailer_preview.rb b/test/mailers/previews/booking_confirmation_mailer_preview.rb
deleted file mode 100644
index 6087296..0000000
--- a/test/mailers/previews/booking_confirmation_mailer_preview.rb
+++ /dev/null
@@ -1,4 +0,0 @@
-# Preview all emails at http://localhost:3000/rails/mailers/booking_confirmation_mailer
-class BookingConfirmationMailerPreview < ActionMailer::Preview
-
-end
